pkg/splunk/enterprise/util.go

```diff
@@ -33,10 +34,69 @@ import (
 // kubernetes logger used by splunk.enterprise package
 var log = logf.Log.WithName("splunk.enterprise")
 
+// ApplyCommonSecretObject creates/updates the namespace scoped "splunk-secrets" K8S secret object
+func ApplyCommonSecretObject(client splcommon.ControllerClient, namespace string) (*corev1.Secret, error) {
+	var current corev1.Secret
+
+	// Types of Splunk secret tokens
+	secretTokenTypes := []string{"hec_token", "password", "pass4symmkey", "idxc_secret", "shc_secret"}
+
+	// Check if a K8S secrets object "splunk-secrets" exists in the namespace
+	namespacedName := types.NamespacedName{Namespace: namespace, Name: commonSecretName}
+	err := client.Get(context.TODO(), namespacedName, &current)
+	if err != nil {
+		// Not found, generate values for all types of tokens
+		secretData := make(map[string][]byte)
+		for _, tokenType := range secretTokenTypes {
+			secretData[tokenType] = splcommon.GenerateSecret(secretBytes, 24)
+		}
+
+		current := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      commonSecretName,
+				Namespace: namespace,
+			},
+			Data: secretData,
+		}
+
+		// Create the secret
+		err = splctrl.CreateResource(client, &current)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Found, generate values for only missing types of tokens them
+		var updateNeeded bool = false
+		for _, tokenType := range secretTokenTypes {
+			if _, ok := current.Data[tokenType]; !ok {
+				// Value for token not found, generate
+				current.Data[tokenType] = splcommon.GenerateSecret(secretBytes, 24)
```

Comment:
note again special case for hec_token

Reply:
Fixed in the new commit 9a807f8cbe5f48b760a5123c35f5816d437f1752